### ISSUE TITLE
Zero padding and the sign have the wrong order in string formatting

### ISSUE THREAD

**Issue by buildersteven4** [godotengine/godot#7602]
**Operating system or device - Godot version:**
Windows 7, 

**Issue description:**
The zero padding together with signs behave different than I'm use to. The zero's are added before the sign of the number. So while I would expect the code below to print
+01
-02
-03
it actually prints:
0+1
0-2
0-3

I'd say this is undesirable behaviour.

**Steps to reproduce:**
```
func _ready():
	print("%+03d" % 1)
	print("%+03d" % -2)
	print("%03d" % -3)
```

**Comment by xyroclast** [comment]
Can confirm that the issue is present on Linux Mint 18 as well, exactly as described above.

**Comment by razvanc-r** [comment]
Why do most people not include the Godot version?! This never happened to me (from v2.0 to current v2.1.2) on Ubuntu.

**Comment by bojidar-bg** [comment]
Looking at the code, it seems like that's exactly what is "supposed" to happen.

**Comment by akien-mga** [comment]
Fixed by #7756.
